tests/t135_trigger_time2.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+
+from runtest import TestBase
+import subprocess as sp
+
+TDIR='xxx'
+START=0
+
+class TestCase(TestBase):
+    def __init__(self):
+        TestBase.__init__(self, 'sleep', """
+# DURATION    TID     FUNCTION
+            [27437] | main() {
+            [27437] |   foo() {
+   2.241 us [27437] |     mem_alloc();
+            [27437] |     bar() {
+   2.183 ms [27437] |       usleep();
+   2.185 ms [27437] |     } /* bar */
+            [27437] |     mem_free() {
+   3.086 us [27437] |       free();
+   3.806 us [27437] |     } /* mem_free */
+   2.191 ms [27437] |   } /* foo */
+   2.192 ms [27437] | } /* main */
+""", sort='simple')
+
+    def pre(self):
+        global START
+
+        record_cmd = '%s record -F main -d %s %s' % (TestBase.ftrace, TDIR, 't-' + self.name)
+        sp.call(record_cmd.split())
+
+        # find timestamp of function 'malloc'
+        replay_cmd = '%s replay -d %s -F malloc' % (TestBase.ftrace, TDIR)
+        p = sp.Popen(replay_cmd, shell=True, stdout=sp.PIPE, stderr=sp.PIPE)
+        r = p.communicate()[0].decode()
+        START, unit = r.split('\n')[1].split()[0:2] # skip header
+        START = str(float(START) + 0.001) # for time filtering
+        START += unit
+        p.wait()
+
+        return TestBase.TEST_SUCCESS
+
+    def runcmd(self):
+        return '%s replay -T main@time=%s -d %s' % (TestBase.ftrace, START, TDIR)
```

Comment:
Here the time filter value might be longer than 3 digits.  For my (failed) test case, it was `-T main@time=1.2799999999999998us`  so I think it's better to print it as float with a precision limit:

```
diff --git a/tests/t135_trigger_time2.py b/tests/t135_trigger_time2.py
index f16230d..268f844 100644
--- a/tests/t135_trigger_time2.py
+++ b/tests/t135_trigger_time2.py
@@ -4,7 +4,8 @@ from runtest import TestBase
 import subprocess as sp
 
 TDIR='xxx'
-START=0
+TIME=0
+UNIT=''
 
 class TestCase(TestBase):
     def __init__(self):
@@ -24,7 +25,7 @@ class TestCase(TestBase):
 """, sort='simple')
 
     def pre(self):
-        global START
+        global TIME, UNIT
 
         record_cmd = '%s record -F main -d %s %s' % (TestBase.ftrace, TDIR, 't-' + self.name)
         sp.call(record_cmd.split())
@@ -33,15 +34,14 @@ class TestCase(TestBase):
         replay_cmd = '%s replay -d %s -F malloc' % (TestBase.ftrace, TDIR)
         p = sp.Popen(replay_cmd, shell=True, stdout=sp.PIPE, stderr=sp.PIPE)
         r = p.communicate()[0].decode()
-        START, unit = r.split('\n')[1].split()[0:2] # skip header
-        START = str(float(START) + 0.001) # for time filtering
-        START += unit
+        TIME, UNIT = r.split('\n')[1].split()[0:2] # skip header
+        TIME = float(TIME) + 0.001 # for time filtering
         p.wait()
 
         return TestBase.TEST_SUCCESS
 
     def runcmd(self):
-        return '%s replay -T main@time=%s -d %s' % (TestBase.ftrace, START, TDIR)
+        return '%s replay -T main@time=%.3f%s -d %s' % (TestBase.ftrace, TIME, UNIT, TDIR)
 
     def post(self, ret):
         sp.call(['rm', '-rf', TDIR])
```